packages/medusa/src/services/cart.ts

```diff
@@ -2006,31 +2068,36 @@ class CartService extends TransactionBaseService {
       const lineItemServiceTx =
         this.lineItemService_.withTransaction(transactionManager)
 
+      console.log(cart)
       cart.items = (
         await Promise.all(
           cart.items.map(async (item) => {
-            const availablePrice = await this.priceSelectionStrategy_
-              .withTransaction(transactionManager)
-              .calculateVariantPrice(item.variant_id, {
-                region_id: region.id,
-                currency_code: region.currency_code,
-                quantity: item.quantity,
-                customer_id: customer_id || cart.customer_id,
-                include_discount_prices: true,
-              })
-              .catch(() => undefined)
+            if (item.variant_id) {
+              const availablePrice = await this.priceSelectionStrategy_
+                .withTransaction(transactionManager)
+                .calculateVariantPrice(item.variant_id, {
+                  region_id: region.id,
+                  currency_code: region.currency_code,
+                  quantity: item.quantity,
+                  customer_id: customer_id || cart.customer_id,
+                  include_discount_prices: true,
+                })
+                .catch(() => undefined)
 
-            if (
-              availablePrice !== undefined &&
-              availablePrice.calculatedPrice !== null
-            ) {
-              return lineItemServiceTx.update(item.id, {
-                has_shipping: false,
-                unit_price: availablePrice.calculatedPrice,
-              })
+              if (
+                availablePrice !== undefined &&
+                availablePrice.calculatedPrice !== null
+              ) {
+                return lineItemServiceTx.update(item.id, {
```

Comment:
todo: Add missing await
```suggestion
                return await lineItemServiceTx.update(item.id, {
```